src/bam/collapse_reads_to_fragments/pipeline.rs

```diff
@@ -86,82 +140,110 @@ impl<W: io::Write> CallConsensusRead<W> {
+                                    {
+                                        self.bam_skipped_writer.write(r1_rec)?;
+                                        self.bam_skipped_writer.write(&record)?;
+                                        None
+                                    } else {
+                                        duplicate_groups
+                                            .entry(GroupId::Regular(duplicate_id))
+                                            .or_insert_with(Vec::new)
+                                            .push(RecordId::Regular(record_name.to_owned()));
+                                        r2_rec.get_or_insert(IndexedRecord {
+                                            rec: record,
+                                            rec_id: i,
+                                        });
+                                        Some(GroupId::Regular(duplicate_id))
+                                    };
+                                    group_id
                                 }
                                 // This arm is reached if a mate is mapped to another chromosome.
                                 // In that case a new duplicate and record ID is required
-                                RecordStorage::SingleRecord { .. } => {
-                                    let group_id = duplicate_id;
+                                RecordStorage::SingleRecord { rec } => {
+                                    let group_id = if cigar_has_softclips(rec)
+                                        || cigar_has_softclips(&record)
+                                    {
+                                        self.bam_skipped_writer.write(rec)?;
+                                        self.bam_skipped_writer.write(&record)?;
+                                        None
+                                    } else {
+                                        duplicate_groups
+                                            .entry(GroupId::Split(duplicate_id))
+                                            .or_insert_with(Vec::new)
+                                            .push(RecordId::Split(record_name.to_owned()));
+                                        record_storage.insert(
+                                            RecordId::Split(record_name.to_owned()),
+                                            RecordStorage::SingleRecord {
+                                                rec: IndexedRecord {
+                                                    rec: record,
+                                                    rec_id: i,
+                                                },
+                                            },
+                                        );
+                                        Some(GroupId::Split(duplicate_id))
+                                    };
+                                    group_id
+                                }
```

Comment:
Also looks like a code duplication with above. Either delegate to function or use traits.

Reply:
There is not that much code duplicates. Effectively, the if case where `if`-arm where records are written to the bam writer.
The `else`-arms have different logic. I could move the whole `match`-closure into a separate function. But this would also separate the nesting-structure making it harder to track what kind of read constellation is being handled and also require to pass a lot of variables to that function.

Reply:
ok, let's keep it like this.